addons/sale_loyalty/models/sale_order.py

```diff
@@ -592,6 +618,55 @@ def _write_vals_from_reward_vals(self, reward_vals, old_lines, delete=True):
         self.write({'order_line': command_list})
         return self.env['sale.order.line'] if delete else old_lines[len(reward_vals):]
 
+
+    def _is_better_global_discount_applied(self, current_reward, new_reward, discountable=None):
+        """
+        Determines if a better global discount is already applied.
+
+        This function compares the discount amount of the current reward applied to the sale order
+        with the discount amount of a new reward to determine if the new reward provides an identical
+        or higher discount.
+
+        :param loyalty.reward current_reward: The reward currently applied on the sale order.
+        :param loyalty.reward new_reward: The reward to compare with.
+        :return: True if an identical or higher global discount is applied, False otherwise.
+        :rtype: bool
+        """
+        assert current_reward
+
+        def compute_discount(reward, discountable):
+            """
+            Compute the discount amount for the given reward.
+
+            This function computes the discount amount based on the discount mode of the reward:
+            - If the discount mode is 'per_order',
+                it converts the discount into the currency of the sale order on the current date.
+            - If the discount mode is 'percent',
+                it computes the discount amount using `_discountable_order`.
+
+            :param loyalty.reward reward: The reward for which to calculate the maximum discount.
+            :return: The maximum discount amount.
+            :rtype: float
+            """
+            if reward.discount_mode == 'per_order':
+                return reward.currency_id._convert(reward.discount, self.currency_id, self.company_id, fields.Date.today())
+            elif reward.discount_mode == 'percent':
+                return discountable * (reward.discount / 100)
```

Comment:
This is something that could be useful to move on the reward model IMHO (and would reduce the code depth here)